app/src/main/java/study_dev/testbed/studyhopper/StudyGroupActivity.java

```diff
@@ -0,0 +1,135 @@
+package study_dev.testbed.studyhopper;
+
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.cardview.widget.CardView;
+import androidx.core.content.ContextCompat;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.github.mikephil.charting.charts.LineChart;
+import com.github.mikephil.charting.data.Entry;
+import com.github.mikephil.charting.data.LineData;
+import com.github.mikephil.charting.data.LineDataSet;
+import com.github.mikephil.charting.interfaces.datasets.ILineDataSet;
+import com.github.mikephil.charting.listener.OnChartGestureListener;
+import com.github.mikephil.charting.listener.OnChartValueSelectedListener;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+import java.util.ArrayList;
+
+public class StudyGroupActivity extends AppCompatActivity {
+
+    private static final String TAG = "StudyGroupActivity";
+
+    private LineChart mChart;
+    CardView groupMembersCard;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_study_group);
+
+        Intent intent = getIntent();
+        studyGroupItem studyGroupItem = intent.getParcelableExtra("Study Group Item");
+
+        int imageRes = studyGroupItem.getImageResource();
+        String line1 = studyGroupItem.getText1();
+        String line2 = studyGroupItem.getText2();
+
+        ImageView imageView = findViewById(R.id.groupCircleImage);
+        imageView.setImageResource(imageRes);
+
+        TextView textView1 = findViewById(R.id.groupName);
+        textView1.setText(line1);
+
+        TextView textView2 = findViewById(R.id.groupCourseCode);
+        textView2.setText(line2);
+
+        // Enable back button
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar != null) {
+            supportActionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        // Floating action button
+        FloatingActionButton fab = findViewById(R.id.messageFAB);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(StudyGroupActivity.this, Messages.class);
+                startActivity(intent);
```

Comment:
when you go back on messages it goes to dashboard. Shouldn't it go back to StudyGroupActivities

Reply:
Yeah it needs to. I will fix the androidmanifest.xml file to resolve the parent activity.